flake.nix

```diff
@@ -0,0 +1,189 @@
+          exec x86_64-unknown-linux-musl-cc @<(printf "%q\n" "''${newParams[@]}")
+        '';
+
+        # haskell.nix package set
+        # parameters:
+        # - release -- 'true' for master and producion branches builds, 'false' for all other builds.
+        #   This flag basically disables weeder related files production, haddock and enables stripping
+        # - static -- build statically linked executables
+        # - commitSha, commitDate -- git revision info used during compilation of packages with autodoc
+        # - optimize -- 'true' to enable '-O1' ghc flag, we use it for publishing docker image for morley
+        hs-pkgs = { release, static ? true, optimize ? false, commitSha ? null, commitDate ? null}:
+          let
+            haskell-nix = if static then pkgsStatic.haskell-nix else pkgs.haskell-nix;
+          in haskell-nix.stackProject {
+            src = projectSrc;
+
+            # use .cabal files for building because:
+            # 1. haskell.nix fails to work for package.yaml with includes from the parent directory
+            # 2. with .cabal files haskell.nix filters out source files for each component, so only the changed components will rebuild
+            ignorePackageYaml = true;
+
+            modules = [
+              # common options for all local packages:
+              {
+                packages = pkgs.lib.genAttrs local-packages-names (packageName: {
+                  ghcOptions = with pkgs.lib;
+                    # we use O1 for production binaries in order to improve their performance
+                    # for end-users
+                    [ (if optimize then "-O1" else "-O0") "-Werror"]
+                    # override linker to work around issues with Template Haskell
+                    ++ optionals static [ "-pgml=${linker-workaround pkgsStatic}" ]
+                    # produce *.dump-hi files, required for weeder:
+                    ++ optionals (!release) ["-ddump-to-file" "-ddump-hi"]
+                    # do not erase any 'assert' calls
+                    ++ optionals (!release) ["-fno-ignore-asserts"];
+                  dontStrip = !release;  # strip in release mode, reduces closure size
+                  doHaddock = !release;  # don't haddock in release mode
+
+                  # in non-release mode collect all *.dump-hi files (required for weeder)
+                  postInstall = if release then null else weeder-hacks.collect-dump-hi-files;
+                  preBuild = ''
+                    export MORLEY_DOC_GIT_COMMIT_SHA=${if release then pkgs.lib.escapeShellArg commitSha else "UNSPECIFIED"}
+                    export MORLEY_DOC_GIT_COMMIT_DATE=${if release then pkgs.lib.escapeShellArg commitDate else "UNSPECIFIED"}
+                  '';
+                });
+              }
+
+              {
+                # don't haddock dependencies
+                doHaddock = false;
+              }
+            ];
+        };
```

Comment:
Not directly related to this MR, but I'm pretty sure this whole setup (along with some other things, actually) is duplicated across most if not all of our repos. Do you suppose this could also live in morley-infra?